pkg/rpcclient/wsclient.go

```diff
@@ -559,3 +559,82 @@ func (c *WSClient) GetError() error {
 	}
 	return c.closeErr
 }
+
+// Wait waits until transaction is accepted to the chain and returns its execution result
+// or an error if it's missing from chain after VUB block.
+func (c *WSClient) Wait(h util.Uint256, vub uint32) (res *state.AppExecResult, waitErr error) {
+	var wsWaitErr error
+	defer func() {
+		if wsWaitErr != nil {
+			res, waitErr = c.Client.Wait(h, vub)
+			if waitErr != nil {
+				waitErr = fmt.Errorf("WS client error: %w, client error: %v", wsWaitErr, waitErr)
+			}
+		}
+	}()
+	rcvr := make(chan Notification)
+	defer func() {
+	drainLoop:
+		// Drain rcvr to avoid other notification receivers blocking.
+		for {
+			select {
+			case <-rcvr:
+			default:
+				break drainLoop
+			}
+		}
+		close(rcvr)
+	}()
+	blocksID, err := c.subscribeForNewBlocksInternal(nil, rcvr)
+	if err != nil {
+		wsWaitErr = fmt.Errorf("failed to subscribe for new blocks: %w", err)
```

Comment:
Maybe that's where jumping to non-ws waiter would be more appropriate, there is `maxFeeds` limit server-side and it's not that huge.

Reply:
But it was already implemented like that. Please, check the code one more time, I haven't change it.